x-pack/plugins/security_solution/server/endpoint/routes/actions/audit_log.ts

```diff
@@ -28,6 +28,7 @@ export function registerActionAuditLogRoutes(
       access: 'public',
       path: ENDPOINT_ACTION_LOG_ROUTE,
       options: { authRequired: true, tags: ['access:securitySolution'] },
+      deprecated: true,
```

Comment:
`deprecated` needs to be declared on `.addVersion` because the endpoint is being removed completely and needs to be surfaced in the Upgrade Assistant.
```
**
 * Registers the endpoint activity_log route
 * @deprecated
 * @removeBy 9.0.0
 *
 */
export function registerActionAuditLogRoutes(
  router: SecuritySolutionPluginRouter,
  endpointContext: EndpointAppContext
) {
  router.versioned
    .get({
      access: 'public',
      path: ENDPOINT_ACTION_LOG_ROUTE,
      options: { authRequired: true, tags: ['access:securitySolution'] },
    })
    .addVersion(
      {
        version: '2023-10-31',
        options: {
          deprecated: {
            documentationUrl: 'https://elastic.co', // add a link to your docs about the change
            severity: 'critical',
            reason: {
              type: 'remove',
            },
          },
        },
        validate: {
          request: EndpointActionLogRequestSchema,
        },
      },
      withEndpointAuthz(
        { all: ['canIsolateHost'] },
        endpointContext.logFactory.get('hostIsolationLogs'),
        auditLogRequestHandler(endpointContext)
      )
    );
}
```